### PR TITLE
[CoronaVirus] add caching module issue #761

### PR DESCRIPTION
**Pull request by sadeemsaleh**
- when a query is in the cache, the result would be returned from the cache after checking its freshness
- the cache is cleaned whenever we hit the max threshold of some number of queries.
- every thread would have its local copy to maintain the cache until it receives the done.
- once the done message is received, the local cache would push the changes to the central cache.
- whenever two threads try to add to the central cache, it would take the latest timestamp.
- the timestamp is created upon request of the query not the result of the query.

### CODE REVIEW

**Comment by Yicong-Huang** [comment]
resolves #761

File: cloudberry/zion/src/main/scala/edu/uci/ics/cloudberry/zion/actor/RESTSolver.scala [hunk 1]
```diff
@@ -33,5 +33,4 @@ class RESTSolver(val dataManager: ActorRef,
       }
   }
 
-}
-
+}
```
Comment (baiqiushi) [inline cloudberry/zion/src/main/scala/edu/uci/ics/cloudberry/zion/actor/RESTSolver.scala]: Let's avoid unnecessary changes.

File: examples/coronavirustwittermap/web/app/websocket/TwitterMapServerToCloudBerrySocket.scala [hunk 2]
```diff
@@ -68,24 +84,24 @@ class TwitterMapServerToCloudBerrySocket(out: ActorRef, config: Configuration) {
                 /**
                   * 1) pinMapResult response from Cloudberry has format as follows,
                   * { id: "defaultID",
-                  *   category: "pinMapResult",
-                  *   value: [[
-                  *     { id: 1236910669389131776, place.bounding_box: [[-120.582586,46.958017], [-120.49726,47.028542]] },
-                  *     { id: 1236509727560880128, coordinate: [-121.944442,37.325656], place.bounding_box: [[...]]},
-                  *     ... ...
-                  *   ]]
+                  * category: "pinMapResult",
```
Comment (baiqiushi) [inline examples/coronavirustwittermap/web/app/websocket/TwitterMapServerToCloudBerrySocket.scala]: Let's avoid this unnecessary format changes, and also this indent is useful for readers.

Comment (sadeemsaleh) [reply examples/coronavirustwittermap/web/app/websocket/TwitterMapServerToCloudBerrySocket.scala]: Sorry, I didn’t change this part must have been due to reformat code option
